magma/lib/magma/gnomon/grammar.rb

```diff
@@ -109,6 +109,7 @@ def decompose(identifier)
 
         {
           tokens: tokens,
+          rule_name: rule.name,
```

Comment:
Just noting this line for @graft.  There was a new merge conflict here this morning when I went to merge and I had to add the `rule.` portion to this line to make it work with the updated methodology.  I'm pretty that I got it correct because the target rule is still prioritized in <idTreeTable>s, but still wanted to get your eyes on it!